core/kits/assets/js/component.js

```diff
@@ -57,7 +58,12 @@ export default class extends $e.modules.ComponentBase {
 		return this.importHooks( hooks );
 	}
 
-	renderTab( tab ) {
-		elementor.getPanelView().setPage( 'kit_settings' ).content.currentView.activateTab( tab );
+	renderTab( tab, args ) {
+		if ( tab !== this.currentTab ) { // Prevent re-rendering the same tab (with just different args).
+			this.currentTab = tab;
+			elementor.getPanelView().setPage( 'kit_settings' ).content.currentView.activateTab( tab );
+		}
```

Comment:
Why is it needed?

Reply:
To: Prevent re-rendering the same tab (with just different args). Meaning that if the `reRoute` is called and a new activeControl is added to the currentRoute. The route will pass the usually validation checks that check if we are on the same route because the arguments are different, but since it is still the same route we don't want here in the activateTab to rerender everything we just want to handle any new arguments like the `activeControl`